internal/service/docdb/sweep.go

```diff
@@ -23,6 +23,212 @@ func init() {
+		Name:         "aws_docdb_cluster_parameter_group",
+		F:            sweepDBClusterParameterGroups,
+		Dependencies: []string{},
+	})
+}
+
+func sweepDBClusters(region string) error {
+	client, err := sweep.SharedRegionalSweepClient(region)
+
+	if err != nil {
+		return fmt.Errorf("error getting client: %w", err)
+	}
+
+	conn := client.(*conns.AWSClient).DocDBConn
+	input := &docdb.DescribeDBClustersInput{}
+
+	err = conn.DescribeDBClustersPages(input, func(out *docdb.DescribeDBClustersOutput, lastPage bool) bool {
+		for _, dBCluster := range out.DBClusters {
+			id := aws.StringValue(dBCluster.DBClusterIdentifier)
+			input := &docdb.DeleteDBClusterInput{
+				DBClusterIdentifier: dBCluster.DBClusterIdentifier,
+			}
+
+			log.Printf("[INFO] Deleting DocDB Cluster: %s", id)
+
+			_, err := conn.DeleteDBCluster(input)
+
+			if err != nil {
+				log.Printf("[ERROR] Failed to delete DocDB Cluster (%s): %s", id, err)
+				continue
+			}
+
+			if err := WaitForDBClusterDeletion(context.TODO(), conn, id, DBClusterDeleteTimeout); err != nil {
+				log.Printf("[ERROR] Failure while waiting for DocDB Cluster (%s) to be deleted: %s", id, err)
+			}
+		}
+		return !lastPage
+	})
+
+	if sweep.SkipSweepError(err) {
+		log.Printf("[WARN] Skipping DocDB Cluster sweep for %s: %s", region, err)
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error retrieving DocDB Clusters: %w", err)
+	}
+
+	return nil
+}
+
+func sweepDBClusterSnapshots(region string) error {
+	client, err := sweep.SharedRegionalSweepClient(region)
+
+	if err != nil {
+		return fmt.Errorf("error getting client: %w", err)
+	}
+
+	conn := client.(*conns.AWSClient).DocDBConn
+	input := &docdb.DescribeDBClusterSnapshotsInput{}
+
+	err = conn.DescribeDBClusterSnapshotsPages(input, func(out *docdb.DescribeDBClusterSnapshotsOutput, lastPage bool) bool {
+		for _, dBClusterSnapshot := range out.DBClusterSnapshots {
+			name := aws.StringValue(dBClusterSnapshot.DBClusterSnapshotIdentifier)
+			input := &docdb.DeleteDBClusterSnapshotInput{
+				DBClusterSnapshotIdentifier: dBClusterSnapshot.DBClusterSnapshotIdentifier,
+			}
+
+			log.Printf("[INFO] Deleting DocDB Cluster Snapshot: %s", name)
+
+			_, err := conn.DeleteDBClusterSnapshot(input)
+
+			if err != nil {
+				log.Printf("[ERROR] Failed to delete DocDB Cluster Snapshot (%s): %s", name, err)
+				continue
+			}
+
+			if err := WaitForDBClusterSnapshotDeletion(context.TODO(), conn, name, DBClusterSnapshotDeleteTimeout); err != nil {
+				log.Printf("[ERROR] Failure while waiting for DocDB Cluster Snapshot (%s) to be deleted: %s", name, err)
+			}
+		}
+		return !lastPage
+	})
+
+	if sweep.SkipSweepError(err) {
+		log.Printf("[WARN] Skipping DocDB Cluster Snapshot sweep for %s: %s", region, err)
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error retrieving DocDB Cluster Snapshots: %w", err)
+	}
+
+	return nil
+}
+
+func sweepDBClusterParameterGroups(region string) error {
```

Comment:
The only resource that didn't have a status as part of it's object. There's a ParameterGroupStatus, which is only available on a DocDBCluster - I believe that's just whether it's associated with the cluster or whatever. It seems like the ParametGroup resources are deleted instantly without some status associated to the resource..